docs: Improve flat config ignores docs (#17997)
eslint/eslint: docs/src/use/configure/migration-guide.md

];
```

In `.eslintignore`, `temp.js` ignores all files named `temp.js`, whereas in flat config, you need to specify this as `**/temp.js`. The pattern `temp.js` in flat config only ignores a file named `temp.js` in the same directory as the configuration file.

> ::: important
> In flat config , dotfiles (e.g. `.dotfile.js`) are no longer ignored by default. If you want to ignore dotfiles, add an ignore pattern of `"**/.*"`.
> :::

Reviewer: Not sure if the ::: adds too much value in this file (not very familiar with documentation practices of the project)

Below is a suggestion that looks clearer when previewing the MD at least for me.

```suggestion
*important*: When using flat config, dotfiles (e.g: `.dotfile.js`, `.env`) are not longer ignored by default. To ignore dotfiles use an ignore pattern of `"**/*`
```

Alternatively, it seems that your intention was to have a breakline from "important" from the text that clarify what is important, if that is the case you might need to fix the markup.

Review how this is looks now:
<img width="1036" alt="Screenshot 2024-01-19 at 13 51 03" src="https://github.com/eslint/eslint/assets/8984618/5c6c1d05-e9df-4468-a56a-d5c4002bba10">

Then perhaps this is the intention.
```
::: important

In flat config , dotfiles (e.g. `.dotfile.js`) are no longer ignored by default. If you want to ignore dotfiles, add an ignore pattern of `"**/.*"`.

:::

Reviewer 2: > Not sure if the ::: adds too much value in this file (not very familiar with documentation practices of the project)
> 
> Below is a suggestion that looks clearer when previewing the MD at least for me.

MD files are transformed into html for the eslint.org website. You can check how this looks like on the website here:

https://deploy-preview-17997--docs-eslint.netlify.app/use/configure/migration-guide

![image](https://github.com/eslint/eslint/assets/44349756/9a25db78-21a7-4858-baf3-0a20b67569a7)

Reviewer: Oh I see thanks for the feedback
